src/ng/filter/filter.js

```diff
@@ -127,7 +127,7 @@
  */
 function filterFilter() {
   return function(array, expression, comparator) {
-    if (!isArray(array)) {
+    if (!isArrayLike(array)) {
```

Comment:
Because of my poor english I normally don't write down my arguments too well. 
I'll try to explain my point a little more (however, and of course, I don't have any problem to change the PR to make the check).

In the jsperf, we see very little difference given between `array with check` and `array without check` that:
* we have very few elements
* most important, the filterFn is super simple and gets superoptimized by the compiler

In a more realistic scenario where we may be concerned by performance, I think we'll have this differences:
* the number of elements will be two orders of magnitude bigger or more
* the `predicateFn` will be one created by `createPredicateFn`, that will be way slower than the one optimized in the jsbin

By this reasons, IMHO, I think that the improvement of the check is so small when performance is an issue, that is not worth it.

But again, if you think it is worth it, I'll change the PR :)